content/Sessions/Sessions.q.md

```diff
@@ -0,0 +1,924 @@
+?>
+```
+/// type=MS, answer=[3,4,5]
+
+Why does adding the statement `unset($_SESSION["firstName"]);` produce an error?
+
+ - On line 7, the key `firstName` is enclosed in double quotes `""`. 
+
+ - On line 7, the `unset($_SESSION["firstName"]);` statement is invalid.
+
+ - The `$_SESSION["firstName"]` session variable is no longer accessible on line 8.
+
+ - The session data of the `$_SESSION["firstName"]` session variable does not exists.
+
+ - On line 7, the `unset($_SESSION["firstName"]);` statement removes the `$_SESSION["firstName"]` session variable.
+
+
+/// type=SS, answer=[1]
+
+What does `unset($_SESSION["firstName"]);` do?
+
+ - It deletes the `$_SESSION["firstName"]` session variable.
+
+ - It defines the `$_SESSION["firstName"]` session variable.
+
+ - It returns the `$_SESSION["firstName"]` session variable.
+
+ - It displays the `$_SESSION["firstName"]` session variable.
+
+ - It accesses the `$_SESSION["firstName"]` session variable.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+session_start();
+
+if (!empty($_POST)) {
+    if (isset($_POST['username']) && isset($_POST['password'])) {
+        echo "Hello, " . $_POST['username'] . '.';
+    }
+}
+?>
+
+<html>
+    <body>
+        <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+        </form>
+    </body>
+</html>
+```
+/// type=SS, choiceType=HTML, answer=[5]
+
+Execute the program. What is its output?
+
+ - <form action="" method="post">
+            <input type="submit" value="Submit">
+            <input type="password" name="password" placeholder="Password" required> 
+    </form>
+
+ - <form action="" method="post">
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="submit" value="Submit">
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+    </form>
+
+
+/// type=SS, answer=[5]
+
+In the browser window, click the `Submit` button. What is its result?
+
+ - It prints `Hello`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `Hello, user.`.
+
+ - It displays a message `This is a required field`.
```

Comment:
We do not need to specify this since this will be displayed in our reader with the created browser like behavior so this is not browser dependent. This is similar to what we have in CSS and HTML.

Reply:
Noted. Thank you.